slack_bolt/request/internals.py

```diff
@@ -208,6 +208,26 @@ def extract_channel_id(payload: Dict[str, Any]) -> Optional[str]:
     return None
 
 
+def extract_function_execution_id(payload: Dict[str, Any]) -> Optional[str]:
+    if payload.get("function_execution_id") is not None:
+        return payload.get("function_execution_id")
+    if payload.get("event") is not None:
+        return extract_function_execution_id(payload["event"])
+    if payload.get("function_data") is not None:
+        return extract_function_execution_id(payload["function_data"])
+    if payload.get("execution_id") is not None:
+        return payload.get("execution_id")
```

Comment:
The more general term "execution_id" can be used in a different menaing in the future (say, as part of an Events API payload). So avoiding the recursive calls for it would be safer:
```suggestion
    if payload.get("function_data") is not None:
        return payload["function_data"].get("execution_id")
```